gateway/src/apicast/policy/jwt_claim_check/jwt_claim_check.lua

```diff
@@ -0,0 +1,100 @@
+local policy = require('apicast.policy')
+local _M = policy.new('JWT check policy')
+
+local Condition = require('apicast.conditions.condition')
+local MappingRule = require('apicast.mapping_rule')
+local Operation = require('apicast.conditions.operation')
+local TemplateString = require('apicast.template_string')
+
+local ipairs = ipairs
+
+local new = _M.new
+
+local default_error_message = "Request block due JWT claim policy"
+local default_template_type = 'plain'
+
+local function deny_request(error_msg)
+  ngx.status = ngx.HTTP_FORBIDDEN
+  ngx.say(error_msg)
+  ngx.exit(ngx.status)
+end
+
+function _M.new(config)
+  local self = new(config)
+  self.error_message = config.error_message or default_error_message
+  self.rules = {}
+  for _, rule in ipairs(config.rules) do
+    local conditions = {}
+    for _, condition in ipairs(rule.operations) do
+      if condition.jwt_claim_type == "plain" then
+        -- Due to this need to be fetched from the JWT claim, render  the match
+        -- as liquid to be able to fetch the info from the JWT_claim
+        condition.jwt_claim = "{{"..condition.jwt_claim.."}}"
+      end
+
+      table.insert( conditions,
+        Operation.new(
+          condition.jwt_claim, "liquid",
+          condition.op,
+          condition.value, condition.value_type or default_template_type))
+    end
+    table.insert( self.rules, {
+      condition = Condition.new(conditions, rule.combine_op),
+      methods = rule.methods,
+      resource = TemplateString.new(
+        rule.resource,
+        rule.resource_type or default_template_type),
+    })
+  end
+
+  return self
+end
+
+-- is_rule_denied_request returns true if the request need to be blocked based
+-- on a provided rule with the request context.
+-- This function will only work if the request match on resource and in one of
+-- the methods, the methods that are not  defined in the rule will be allowed. 
+local function is_rule_denied_request(rule, context)
+
+  local uri = ngx.var.uri
+  local request_method =  ngx.req.get_method()
+
+  local resource = rule.resource:render(context)
+  local method_scope_match = false
```

Comment:
I think the name of this var is confusing because it does not just refer to the method. It also refers to the path of the request, doesn't it?